fix: Resolve version before checking arch

abhinav opened the pull request:
Fixes a bug where `gimme module` reports the following error
on macOS with ARM64.

    error: arm64 is not supported by this go version
    try go1.5 or newer

This occurs because the check for architecture
is performed before the `module` version
turns into a real version number with `_resolve_version`.

Fix by resolving version number before checking the architecture.

abhinav committed:
fix: Resolve version before checking arch

Fixes a bug where `gimme module` reports the following error
on macOS with ARM64.

    error: arm64 is not supported by this go version
    try go1.5 or newer

This occurs because the check for architecture
is performed before the `module` version
turns into a real version number with `_resolve_version`.

Fix by resolving version number before checking the architecture.

brackendawson commented:
This fix works for me from 1.16 (when darwin/arm64 support was added). But the behaviour if the go directive specifies <1.16  changes from a partially incorrect error message (should say 1.16 on darwin):
```
% gimme module
error: arm64 is not supported by this go version
try go1.5 or newer
```

to an unhelpful error message:
```
% gimme module
I don't have any idea what to do with '1.13.15'.
  (using download type 'auto')
```

Can we restore the original error message but correct the version number for darwin? 1.5 was when linux/arm64 support was added.

brackendawson reviewed (changes requested):
This fix works for me from 1.16 (when darwin/arm64 support was added). But the behaviour if the go directive specifies <1.16  changes from a partially incorrect error message (should say 1.16 on darwin):
```
% gimme module
error: arm64 is not supported by this go version
try go1.5 or newer
```

to an unhelpful error message:
```
% gimme module
I don't have any idea what to do with '1.13.15'.
  (using download type 'auto')
```

Can we restore the original error message but correct the version number for darwin? 1.5 was when linux/arm64 support was added.

brackendawson commented:
Also [this](https://github.com/urfave/gimme/blob/975cf321d374194dae72c273f31979bee67a51bb/README.md?plain=1#L85) needs to change, perhaps:

> To install the release specified in the `go.mod` file, or the most recent patch when no patch version is present:

abhinav committed:
arm64 on darwin is available for go1.16 or newer

Fixes the error message that is printed
if you request Go < 1.16 on darwin/arm64.

abhinav committed:
README: Update `go module` docs following #25

abhinav commented:
Sounds good!
Updated the arm64 version matching.

Agreed about the README change too.
Should've done that in #25, but late is better than never.
Updated to reflect new behavior.

brackendawson commented:
New error message looks good but it still doesn't actually show up:

```
% uname -sm
Darwin arm64
% cat go.mod
module github.com/urfave/gimme/test

go 1.13
% ../gimme module
I don't have any idea what to do with '1.13.15'.
  (using download type 'auto')
```

abhinav commented:
That's weird.

```
❯ uname -sm
Darwin arm64

❯ head -n3 go.mod
module github.com/urfave/gimme

go 1.13

❯ ./gimme module
error: arm64 is not supported by this go version
try go1.16 or newer
```

brackendawson reviewed (approved):
My bad, it works better when you pull all the changes in the branch. Thank you.
